`ajour update` shows no WoW directory is set, when Retail directory is set

**Describe the bug**
ajour update displays:
09:22:55.097 [ajour][INFO] Ajour 1.0.0 has started.
09:22:55.097 [ajour::command::update_addons][INFO] Checking for addon updates...
09:22:56.379 [ajour][ERROR] No WoW directory set. Launch Ajour and make sure a WoW directory is set before using the command line.

**To Reproduce**
Steps to reproduce the behavior:
1. Have an existing ajour configuration or create a new one via the UI, setting only a WoW retail directory.
2. Run `ajour update`

**Expected behavior**
Ajour performs addon updates

**Software involved**
Please complete the following information:
 - OS: Arch Linux 5.11.8-arch1-1
 - Ajour version: Ajour 1.0.0
 - (if applicable) Addon name, flavor (classic or retail) and version: Retail

**Additional context**
ajour.yml
```
---
wow:
  directory: ~
  directories:
    Retail: /home/USERNAME/Games/battlenet/drive_c/Program Files (x86)/World of Warcraft/_retail_
  flavor: Retail
addons:
  global_release_channel: Stable
  ignored: {}
  release_channels: {}
  delete_saved_variables: false
theme: ~
column_config:
  V1:
    local_version_width: 150
    remote_version_width: 150
    status_width: 85
window_size:
  - 1918
  - 1050
scale: ~
backup_directory: ~
backup_addons: false
backup_wtf: false
hide_ignored_addons: false
self_update_channel: Stable
weak_auras_account: {}
alternating_row_colors: false
language: English
catalog_source: ~
auto_update: false
```

**Log Output**
The Log appears only to get written to, when starting the GUI.
```
09:22:30.377 [ajour][INFO] Ajour 1.0.0 has started.
09:22:30.377 [ajour::gui][DEBUG] config loaded:
Config {
    wow: Wow {
        directory: None,
        directories: {},
        flavor: Retail,
    },
    addons: Addons {
        global_release_channel: Stable,
        ignored: {},
        release_channels: {},
        delete_saved_variables: false,
    },
    theme: None,
    column_config: V1 {
        local_version_width: 150,
        remote_version_width: 150,
        status_width: 85,
    },
    window_size: None,
    scale: None,
    backup_directory: None,
    backup_addons: false,
    backup_wtf: false,
    hide_ignored_addons: false,
    self_update_channel: Stable,
    weak_auras_account: {},
    alternating_row_colors: false,
    language: English,
    catalog_source: None,
    auto_update: false,
}
09:22:30.377 [ajour::gui][DEBUG] antialiasing: true
09:22:30.435 [ajour_core::theme][DEBUG] loading user themes
09:22:30.435 [ajour_core::utility][DEBUG] checking for application update
09:22:30.437 [ajour_core::fs::theme][DEBUG] loaded 0 user themes
09:22:30.454 [ajour::gui::update][DEBUG] Message::CachesLoaded(error: false)
09:22:30.454 [ajour::gui::update][DEBUG] Message::ThemesLoaded(0 themes)
09:22:30.456 [ajour::gui::update][DEBUG] Message::Parse
09:22:30.722 [ajour_core::catalog][DEBUG] Downloaded latest catalog from https://github.com/ajour/ajour-catalog/raw/master/catalog-2.0.json
09:22:30.812 [ajour::gui::update][DEBUG] Message::LatestRelease(Some("1.0.0"))
09:22:31.049 [ajour::gui::update][DEBUG] Message::CatalogDownloaded(16024 addons in catalog)
09:22:32.345 [ajour::gui::update][DEBUG] Interaction::SelectWowDirectory(None)
09:22:44.457 [ajour::gui::update][DEBUG] Message::UpdateWowDirectory(Chosen(Some("/home/USERNAME/Games/battlenet/drive_c/Program Files (x86)/World of Warcraft") - None))
09:22:44.457 [ajour::gui::update][DEBUG] Message::UpdateWowDirectory(Resolution("/home/USERNAME/Games/battlenet/drive_c/Program Files (x86)/World of Warcraft"))
09:22:44.458 [ajour::gui::update][DEBUG] Message::Parse
09:22:44.458 [ajour::gui::update][DEBUG] preparing to parse addons in "/home/USERNAME/Games/battlenet/drive_c/Program Files (x86)/World of Warcraft/_retail_/Interface/AddOns"
09:22:44.458 [ajour_core::parse][DEBUG] Retail - parsing addons folder
09:22:44.458 [ajour_core::parse][DEBUG] Retail - 125 folders in AddOns directory to parse
09:22:44.474 [ajour::gui::update][DEBUG] Message::CheckWeakAurasInstalled(Retail, is_installed: true)
09:22:44.491 [ajour::gui::update][DEBUG] Message::ListWeakAurasAccounts(Retail, num_accounts: 1)
09:22:45.557 [ajour::gui::update][DEBUG] Message::ParsedAuras(Retail, num_auras: 38)
09:22:46.358 [ajour_core::parse][DEBUG] Retail - 125 fingerprints: 0 cached, 125 calculated, 125 added, 0 removed
09:22:46.367 [ajour_core::parse][DEBUG] Retail - 125 addon folders successfully parsed from '.toc'
09:22:46.367 [ajour_core::parse][DEBUG] Retail - 0 valid cache entries retrieved
09:22:46.367 [ajour_core::parse][DEBUG] Retail - 125 unique fingerprints to check against curse api
09:22:47.324 [ajour_core::parse][DEBUG] Retail - 69 curse packages fetched
09:22:47.324 [ajour_core::parse][DEBUG] Retail - 0 tukui packages fetched
09:22:47.578 [ajour_core::parse][DEBUG] Retail - 22 wowi packages fetched
09:22:47.578 [ajour_core::parse][DEBUG] Retail - 0 townlong packages fetched
09:22:47.578 [ajour_core::parse][DEBUG] Retail - 0 git packages fetched
09:22:47.578 [ajour_core::parse][DEBUG] Retail - 67 addons built from curse packages
09:22:47.578 [ajour_core::parse][DEBUG] Retail - 0 addons built from tukui packages
09:22:47.578 [ajour_core::parse][DEBUG] Retail - 0 addons built from wowi packages
09:22:47.578 [ajour_core::parse][DEBUG] Retail - 0 addons built from townlong-yak packages
09:22:47.578 [ajour_core::parse][DEBUG] Retail - 0 addons built from git packages
09:22:47.578 [ajour_core::parse][DEBUG] Retail - 2 unknown addon folders
09:22:47.578 [ajour_core::parse][DEBUG] Retail - 69 addons successfully parsed
09:22:47.580 [ajour::gui::update][DEBUG] Message::ParsedAddons(Retail, 69 addons)
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -14,6 +14,10 @@ and `Removed`.
 
 ## [Unreleased]
 
+### Fix
+
+- CLI now works properly with the new multiple path setup.
+
 ## [1.0.0] - 2021-03-23
 
 ### Added

--- a/src/command/backup.rs
+++ b/src/command/backup.rs
@@ -45,12 +45,21 @@ pub fn backup(
         let mut src_folders = vec![];
 
         for flavor in flavors {
-            let wow_dir = config.get_root_directory_for_flavor(&flavor).ok_or_else(|| format_err!("No WoW directories set. Launch Ajour and make sure a WoW directory is set before using the command line."))?;
-            let addon_directory = config.get_addon_directory_for_flavor(&flavor).ok_or_else(|| format_err!("No WoW directories set. Launch Ajour and make sure a WoW directory is set before using the command line."))?;
-            let wtf_directory = config.get_wtf_directory_for_flavor(&flavor).ok_or_else(|| format_err!("No WoW directories set. Launch Ajour and make sure a WoW directory is set before using the command line."))?;
+            let wow_directory = match config.get_root_directory_for_flavor(&flavor) {
+                Some(path) => path,
+                None => continue,
+            };
+            let addon_directory = match config.get_addon_directory_for_flavor(&flavor) {
+                Some(path) => path,
+                None => continue,
+            };
+            let wtf_directory = match config.get_wtf_directory_for_flavor(&flavor) {
+                Some(path) => path,
+                None => continue,
+            };
 
-            let addons_folder = backup::BackupFolder::new(&addon_directory, &wow_dir);
-            let wtf_folder = backup::BackupFolder::new(&wtf_directory, &wow_dir);
+            let addons_folder = backup::BackupFolder::new(&addon_directory, &wow_directory);
+            let wtf_folder = backup::BackupFolder::new(&wtf_directory, &wow_directory);
 
             match backup_folder {
                 BackupFolder::Both => {

--- a/src/command/install.rs
+++ b/src/command/install.rs
@@ -45,8 +45,12 @@ pub fn install_from_source(url: Uri, flavor: Flavor) -> Result<()> {
 
         log::debug!("Installing {} for {:?}", addon.title(), flavor);
 
-        let download_directory = config.get_download_directory_for_flavor(flavor).ok_or_else(|| format_err!("No known WoW directory found for the selected flavor."))?;
-        let addon_directory = config.get_addon_directory_for_flavor(&flavor).ok_or_else(|| format_err!("No known WoW directory found for the selected flavor."))?;
+        let download_directory = config
+            .get_download_directory_for_flavor(flavor)
+            .ok_or_else(|| format_err!("No known WoW directory found for the selected flavor."))?;
+        let addon_directory = config
+            .get_addon_directory_for_flavor(&flavor)
+            .ok_or_else(|| format_err!("No known WoW directory found for the selected flavor."))?;
 
         // Download the addon
         download_addon(&addon, global_release_channel, &download_directory).await?;

--- a/src/command/install.rs
+++ b/src/command/install.rs
@@ -45,8 +45,8 @@ pub fn install_from_source(url: Uri, flavor: Flavor) -> Result<()> {
 
         log::debug!("Installing {} for {:?}", addon.title(), flavor);
 
-        let download_directory = config.get_download_directory_for_flavor(flavor).ok_or_else(|| format_err!("No WoW directory set. Launch Ajour and make sure a WoW directory is set before using the command line."))?;
-        let addon_directory = config.get_addon_directory_for_flavor(&flavor).ok_or_else(|| format_err!("No WoW directory set. Launch Ajour and make sure a WoW directory is set before using the command line."))?;
+        let download_directory = config.get_download_directory_for_flavor(flavor).ok_or_else(|| format_err!("No known WoW directory found for the selected flavor."))?;
+        let addon_directory = config.get_addon_directory_for_flavor(&flavor).ok_or_else(|| format_err!("No known WoW directory found for the selected flavor."))?;
 
         // Download the addon
         download_addon(&addon, global_release_channel, &download_directory).await?;

--- a/src/command/update_addons.rs
+++ b/src/command/update_addons.rs
@@ -14,7 +14,7 @@ use ajour_core::network::download_addon;
 use ajour_core::parse::{read_addon_directory, update_addon_fingerprint};
 use ajour_core::repository::{GlobalReleaseChannel, RepositoryKind};
 
-use anyhow::{format_err, Context};
+use anyhow::Context;
 use async_std::sync::{Arc, Mutex};
 use async_std::task;
 
@@ -38,9 +38,13 @@ pub fn update_all_addons() -> Result<()> {
         let mut addons_to_update = vec![];
 
         // Update addons for both flavors
-        for flavor in Flavor::ALL.iter() {
-            // Only returns None if the path isn't set in the config
-            let addon_directory = config.get_addon_directory_for_flavor(flavor).ok_or_else(|| format_err!("No WoW directory set. Launch Ajour and make sure a WoW directory is set before using the command line."))?;
+        let flavors = config.wow.directories.keys().collect::<Vec<_>>();
+        for flavor in flavors {
+            // Returns None if no path is found
+            let addon_directory = match config.get_addon_directory_for_flavor(flavor) {
+                Some(path) => path,
+                None => continue,
+            };
 
             if let Ok(addons) = read_addon_directory(
                 Some(addon_cache.clone()),

--- a/src/command/update_addons.rs
+++ b/src/command/update_addons.rs
@@ -37,7 +37,7 @@ pub fn update_all_addons() -> Result<()> {
 
         let mut addons_to_update = vec![];
 
-        // Update addons for both flavors
+        // Update addons for known flavors
         let flavors = config.wow.directories.keys().collect::<Vec<_>>();
         for flavor in flavors {
             // Returns None if no path is found

--- a/src/command/update_weakauras.rs
+++ b/src/command/update_weakauras.rs
@@ -1,9 +1,9 @@
 use crate::Result;
 
-use ajour_core::config::{load_config, Flavor};
+use ajour_core::config::load_config;
 use ajour_weak_auras::{get_aura_updates, parse_auras, write_updates};
 
-use anyhow::{bail, format_err, Context};
+use anyhow::{bail, Context};
 use async_std::task;
 
 pub fn update_all_weakauras() -> Result<()> {
@@ -14,14 +14,21 @@ pub fn update_all_weakauras() -> Result<()> {
 
         let mut flavors_setup = 0usize;
 
-        for flavor in &Flavor::ALL[..] {
+        let flavors = config.wow.directories.keys().collect::<Vec<_>>();
+        for flavor in flavors {
             if let Some(account) = config.weak_auras_account.get(flavor).cloned() {
                 flavors_setup += 1;
 
                 log::info!("{} - Parsing WeakAuras for account {}", flavor, &account);
 
-                let wtf_path = config.get_wtf_directory_for_flavor(flavor).ok_or_else(|| format_err!("No WoW directory set. Launch Ajour and make sure a WoW directory is set before using the command line."))?;
-                let addon_dir = config.get_addon_directory_for_flavor(flavor).ok_or_else(|| format_err!("No WoW directory set. Launch Ajour and make sure a WoW directory is set before using the command line."))?;
+                let wtf_path = match config.get_wtf_directory_for_flavor(flavor) {
+                    Some(path) => path,
+                    None => continue,
+                };
+                let addon_dir = match config.get_addon_directory_for_flavor(flavor) {
+                    Some(path) => path,
+                    None => continue,
+                };
 
                 let auras = parse_auras(wtf_path, account.clone())
                     .await

